src/libs/Xinterface/Nodes/xi_questtitles.cpp

```diff
@@ -369,22 +369,21 @@ void CXI_QUESTTITLE::SetNewTopQuest(ATTRIBUTES *pA, int topNum)
             const char *pTmpQuestRecordID = pAttr->GetAttribute("LogName");
             if (!pTmpQuestRecordID)
                 pTmpQuestRecordID = pAttr->GetThisName();
-            if (ptrOwner->QuestFileReader()->GetQuestTitle(pTmpQuestRecordID, pAttr->GetThisName(), sizeof(param) - 1,
-                                                           param))
+            if (ptrOwner->QuestFileReader()->GetQuestTitle(pTmpQuestRecordID, pAttr->GetThisName(), param))
             {
-                const int titleSize = strlen(param);
+                const size_t titleSize = param.size();
                 if (titleSize == 0)
                     m_strList[i].lineQuantity = 0;
                 else
                 {
-                    char lineName[sizeof(param)];
-                    auto *pstr = param;
+                    std::vector<char> lineName(titleSize);
+                    auto *pstr = param.data();
                     int ln = 0;
-                    while (GetLineNext(m_idFont, pstr, lineName, sizeof(lineName)))
+                    while (GetLineNext(m_idFont, pstr, lineName.data(), lineName.size())) /// @todo
```

Comment:
is to-do still valid? maybe add a comment what still needs to be done then?

Reply:
Yeah, forgot to complete a comment